core/src/main/java/brooklyn/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -750,7 +780,36 @@ private String setFromItemIfUnset(String oldValue, Map<?,?> item, String fieldAt
         return oldValue;
     }
 
-    
+    /** scans the given libraries for annotated items, or if null scans the local classpath */ 
```

Comment:
I'd prefer it if `null` does not mean "local scan". In (too) many other places we treat null the same as an empty list. Treating it as meaning "local" will be surprising.

If non-null, does it scan the local classpath as well? I'm finding it hard to tell from the code what triggers it to scan the local classpath.